<!DOCTYPE html>
<html lang="en-us">
<head>

  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="theme" content="hugo-academic">
  <meta name="generator" content="Hugo 0.25.1" />
  <meta name="author" content="Ben Kellman">
  <meta name="description" content="Ph.D. Student in Bioinformatics and Systems Biology">

  
  
  
    
  
  
    
    
    <link rel="stylesheet" href="/css/highlight.min.css">
    
  
  
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha512-6MXa8B6uaO18Hid6blRMetEIoPqHf7Ux1tnyIQdpt9qI5OACx7C+O3IVTr98vwGnlcg0LOLa02i9Y1HpVhlfiw==" crossorigin="anonymous">
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/academicons/1.8.0/css/academicons.min.css" integrity="sha512-GGGNUPDhnG8LEAEDsjqYIQns+Gu8RBs4j5XGlxl7UfRaZBhCCm5jenJkeJL8uPuOXGqgl8/H1gjlWQDRjd3cUQ==" crossorigin="anonymous">
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha512-SfTiTlX6kk+qitfevl/7LibUOeJWlt9rbyDn92a1DqWOw9vWG2MFoays0sgObmWazO5BQPiFucnnEAjpAB+/Sw==" crossorigin="anonymous">
  
  
  <link rel="stylesheet" href="//fonts.googleapis.com/css?family=Lato:400,700%7CMerriweather%7CRoboto+Mono">
  <link rel="stylesheet" href="/css/hugo-academic.css">
  

  
    <script>
      window.ga=window.ga||function(){(ga.q=ga.q||[]).push(arguments)};ga.l=+new Date;
      ga('create', 'UA-103577403-1', 'auto');
      ga('require', 'eventTracker');
      ga('require', 'outboundLinkTracker');
      ga('require', 'urlChangeTracker');
      ga('send', 'pageview');
    </script>
    <script async src="//www.google-analytics.com/analytics.js"></script>
    
    <script async src="https://cdnjs.cloudflare.com/ajax/libs/autotrack/2.4.1/autotrack.js" integrity="sha512-HUmooslVKj4m6OBu0OgzjXXr+QuFYy/k7eLI5jdeEy/F4RSgMn6XRWRGkFi5IFaFgy7uFTkegp3Z0XnJf3Jq+g==" crossorigin="anonymous"></script>
    
  

  <link rel="alternate" href="https://bkellman.github.io/index.xml" type="application/rss+xml" title="Ben Kellman">
  <link rel="feed" href="https://bkellman.github.io/index.xml" type="application/rss+xml" title="Ben Kellman">

  <link rel="icon" type="image/png" href="/img/icon.png">
  <link rel="apple-touch-icon" type="image/png" href="/img/apple-touch-icon.png">

  <link rel="canonical" href="https://bkellman.github.io/post/betweensamplenormalization/">

  

  <title>Examining the basis for between sample normalization | Ben Kellman</title>

</head>
<body id="top" data-spy="scroll" data-target="#navbar-main" data-offset="71">

<nav class="navbar navbar-default navbar-fixed-top" id="navbar-main">
  <div class="container">

    
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse"
              data-target=".navbar-collapse" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="/">Ben Kellman</a>
    </div>

    
    <div class="collapse navbar-collapse">

      
      <ul class="nav navbar-nav navbar-right">
        

        

        <li class="nav-item">
          <a href="/#about">
            
            <span>Home</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/#publications">
            
            <span>Publications</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/#posts">
            
            <span>Posts</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/#projects">
            
            <span>Projects</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/#fun">
            
            <span>Fun</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/#news">
            
            <span>News</span>
          </a>
        </li>

        
        

        

        <li class="nav-item">
          <a href="/#contact">
            
            <span>Contact</span>
          </a>
        </li>

        
        

        
      </ul>

    </div>
  </div>
</nav>


<article class="article" itemscope itemtype="http://schema.org/Article">

  


  <div class="article-container">
    <h1 itemprop="name">Examining the basis for between sample normalization</h1>
    

<div class="article-metadata">

  <span class="article-date">
    <time datetime="2016-04-20 12:00:00 &#43;0000 UTC" itemprop="datePublished">
      Wed, Apr 20, 2016
    </time>
  </span>

  

  
  
  
  <span class="article-tags">
    <i class="fa fa-tags"></i>
    
    <a href="/tags/rnaseq">RNAseq</a
    >, 
    
    <a href="/tags/normalization">Normalization</a
    >, 
    
    <a href="/tags/fpkm">FPKM</a
    >, 
    
    <a href="/tags/bsn">BSN</a
    >, 
    
    <a href="/tags/betweensamplenormalization">BetweenSampleNormalization</a
    >
    
  </span>
  
  

  
  
<div class="share-box" aria-hidden="true">
  <ul class="share">
    <li>
      <a class="facebook"
         href="https://www.facebook.com/sharer.php?u=https%3a%2f%2fbkellman.github.io%2fpost%2fbetweensamplenormalization%2f"
         target="_blank">
        <i class="fa fa-facebook"></i>
      </a>
    </li>
    <li>
      <a class="twitter"
         href="https://twitter.com/intent/tweet?text=Examining%20the%20basis%20for%20between%20sample%20normalization&amp;url=https%3a%2f%2fbkellman.github.io%2fpost%2fbetweensamplenormalization%2f"
         target="_blank">
        <i class="fa fa-twitter"></i>
      </a>
    </li>
    <li>
      <a class="linkedin"
         href="https://www.linkedin.com/shareArticle?mini=true&amp;url=https%3a%2f%2fbkellman.github.io%2fpost%2fbetweensamplenormalization%2f&amp;title=Examining%20the%20basis%20for%20between%20sample%20normalization"
         target="_blank">
        <i class="fa fa-linkedin"></i>
      </a>
    </li>
    <li>
      <a class="weibo"
         href="http://service.weibo.com/share/share.php?url=https%3a%2f%2fbkellman.github.io%2fpost%2fbetweensamplenormalization%2f&amp;title=Examining%20the%20basis%20for%20between%20sample%20normalization"
         target="_blank">
        <i class="fa fa-weibo"></i>
      </a>
    </li>
    <li>
      <a class="email"
         href="mailto:?subject=Examining%20the%20basis%20for%20between%20sample%20normalization&amp;body=https%3a%2f%2fbkellman.github.io%2fpost%2fbetweensamplenormalization%2f">
        <i class="fa fa-envelope"></i>
      </a>
    </li>
  </ul>
</div>


  

</div>

    <div class="article-style" itemprop="articleBody">
      

<h2 id="opinion-currently-under-review-as-of-8-7-2016"><em>opinion currently under-review (as of 8/7/2016)</em></h2>

<p>A friend pointed out to me recently that I should not be using FPKM for between sample comparisons and sent me <a href="https://haroldpimentel.wordpress.com/2014/12/08/in-rna-seq-2-2-between-sample-normalization/" target="_blank"><em>this blog post</em></a>. I had some thoughts about its validity and wanted to write them down.</p>

<p>I believe the example used in this blogpost is not an reasonable. This prompted me to play a bit with the numbers and examine the robustness of FPKM in this respect.</p>

<p>Clearly, FPKM (or library normalized counts) breaks down at 5 genes when 1 has a 20-fold increase in differential expression. That breakdown is indicated by the differential expression induced in all the other, stabely expressed, genes.
I&rsquo;ve retooled the example above in general terms:
Normalized Count Treatment = Count_Gene_i / (Count_Gene_i x StableGenes + Count_Gene_Funky x FunkyGenes)</p>

<p>$$NC_t = C_i / (C_i G_s + C_f G_f)$$
$$NC_t = 5/(5(5)+75(1)) = 0.05$$</p>

<p>The concern is that this number diverges from the previous normalized count though it shouldn&rsquo;t:
$$NC_c = 0.16 = 2/(2(5)+2(1))$$</p>

<p><strong>As was pointed out in the blogplost, $NC_t \not= NC_c$ and that is incorrect.</strong></p>

<p>We can specify the expected Fold-Change for Stable genes as:
 $$E[FC] =  NC_t / NC_c $$
using &ldquo;&lsquo;&rdquo; to denote the shift is observed expression, we can expand Fold change of a stable gene to:</p>

<p>$$E[FC] = \frac{ C_i&rsquo; / (C_i&rsquo;  G_s + C_f&rsquo;  G_f) }{ C_i / (C_i  G_s + C_f  G_f) }$$</p>

<p>in the control condition $C_f = C_i$. So we can rewrite the above fold change as:</p>

<p>$$ = \frac{ C_i&rsquo; / (C_i&rsquo; G_s + C_f&rsquo; G_f) }{ C_i / (C_i G_s+G_f) } $$</p>

<p>$$= \frac{ C_i&rsquo; / (C_i&rsquo; G_s + C_f&rsquo; G_f) } { 1 / (G_s+G_f) } $$</p>

<p>$$= \frac{ C_i&rsquo; }{ (C_i&rsquo; G_s + C_f&rsquo; G_f) }  {  (G_s+G_f) }$$</p>

<p>We can simplify by assuming that the funky genes will have an average of a 20-fold increase in expression. An average fold change of 20 is actually a good indication that there is something wrong with your design but I will get to that in a moment). Therefore $C_f&rsquo; = 20C_i&rsquo;$</p>

<p>$$= \frac{ C_i&rsquo; }{ (C_i&rsquo; G_s + 20 C_i&rsquo; G_f) }  {  (G_s+G_f) }$$</p>

<p>$$= \frac{ 1 }{ ( G_s + 20 G_f) }  {  (G_s+G_f) }$$</p>

<p>$$= \frac{  (G_s+G_f)}{( G_s + 20  G_f) } $$</p>

<p><strong>Actually, an expected or &ldquo;average&rdquo; value for fold change is 1.</strong> There should be some upregulated, some down regulated, but on average, the expected fold-change of significantly differentially expressed genes is 1. That means that, on average, in a well-designed experiment: $E[C_f&rsquo;] = 1C_i&rsquo;$
$$E[FC]= \frac{  (G_s+G_f)}{( G_s + G_f) } = 1$$</p>

<p><strong>Once we correct the irregularly high fold change, the expected Fold-Change for Stable genes $E[FC]= 1$</strong></p>

<p><strong>Let&rsquo;s ignore the issue of the unrealistic fold-change for a moment</strong></p>

<p>We should also consider that $G_f &lt;&lt;&lt; G_s$. While $G_f$ is considered high if it comes close to 1,000 differentially expressed genes, $G_s$ in humans is typically between 20,000 and 150,000 depending on the annotation.
Therefore, we can say that there are 2-4 orders of magnitude seperating $G_f$ and $G_s$. Considering this, we can maintain the non-zero average fold change and approximate the above statement as:</p>

<p>$$= \frac{  G_s+G_f}{ G_s + 20  G_f } $$
$$= \frac{  G_s+\epsilon}{ G_s + 20 \epsilon } $$
$$= \frac{  G_s}{ G_s  } $$</p>

<p>Again, we find that with a realistic ratio of $G_f:G_s$, we expect a fold-change in stable genes close to 1 with a margin of error of 2-4 orders of magnitude.</p>

<p>The problem with this example is it assumes a poorly designed or poorly controlled experiment where the average differential expression is 20-fold. If we, remove this mistake from the example, stable gene differential expression goes to 1 which is correct.</p>

    </div>
  </div>

</article>

<div class="container">
  <nav>
  <ul class="pager">
    

    
  </ul>
</nav>

</div>

<div class="article-container">
  

</div>

<footer class="site-footer">
  <div class="container">
    <p class="powered-by">

      &copy; 2017 Ben Kellman &middot; 

      Powered by the <a href="https://github.com/gcushen/hugo-academic" target="_blank">Academic
      theme</a> for <a href="http://gohugo.io" target="_blank">Hugo</a>.

      <span class="pull-right" aria-hidden="true">
        <a href="#" id="back_to_top">
          <span class="button_icon">
            <i class="fa fa-chevron-up fa-2x"></i>
          </span>
        </a>
      </span>

    </p>
  </div>
</footer>

    
    <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/1.12.4/jquery.min.js" integrity="sha512-jGsMH83oKe9asCpkOVkBnUrDDTp8wl+adkB2D+//JtlxO4SrLoJdhbOysIFQJloQFD+C4Fl1rMsQZF76JjV0eQ==" crossorigin="anonymous"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery.imagesloaded/4.1.2/imagesloaded.pkgd.min.js" integrity="sha512-iHzEu7GbSc705hE2skyH6/AlTpOfBmkx7nUqTLGzPYR+C1tRaItbRlJ7hT/D3YQ9SV0fqLKzp4XY9wKulTBGTw==" crossorigin="anonymous"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/gsap/1.19.1/TweenMax.min.js" integrity="sha512-Z5heTz36xTemt1TbtbfXtTq5lMfYnOkXM2/eWcTTiLU01+Sw4ku1i7vScDc8fWhrP2abz9GQzgKH5NGBLoYlAw==" crossorigin="anonymous"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/gsap/1.19.1/plugins/ScrollToPlugin.min.js" integrity="sha512-CDeU7pRtkPX6XJtF/gcFWlEwyaX7mcAp5sO3VIu/ylsdR74wEw4wmBpD5yYTrmMAiAboi9thyBUr1vXRPA7t0Q==" crossorigin="anonymous"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha512-iztkobsvnjKfAtTNdHkGVjAYTrrtlC7mGp/54c40wowO7LhURYl3gVzzcEqGl/qKXQltJ2HwMrdLcNUdo+N/RQ==" crossorigin="anonymous"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery.isotope/3.0.4/isotope.pkgd.min.js" integrity="sha512-VDBOIlDbuC4VWxGJNmuFRQ0Li0SKkDpmGyuhAG5LTDLd/dJ/S0WMVxriR2Y+CyPL5gzjpN4f/6iqWVBJlht0tQ==" crossorigin="anonymous"></script>
    
    <script src="/js/hugo-academic.js"></script>
    

    
    
      
      <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.9.0/highlight.min.js"></script>

      

      

      <script>hljs.initHighlightingOnLoad();</script>
    

    
    
    <script type="text/x-mathjax-config">
        MathJax.Hub.Config({ tex2jax: { inlineMath: [['$','$'], ['\\(','\\)']] } });
    </script>
    
    <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS_CHTML" integrity="sha512-tOav5w1OjvsSJzePRtt2uQPFwBoHt1VZcUq8l8nm5284LEKE9FSJBQryzMBzHxY5P0zRdNqEcpLIRVYFNgu1jw==" crossorigin="anonymous"></script>
    
    

  </body>
</html>

